libs/ui/src/hooks/use-activity.hook.ts

```diff
@@ -31,50 +32,109 @@ const transformApiData = (
-        activityData.amount = txData.sends[0]?.amount;
-        activityData.tokenId = txData.sends[0]?.token_id;
-        activityData.symbol = getTokenSymbol(txData.sends[0]?.token_id);
-      } else {
-        activityData.tokenId = txData.receives[0]?.token_id;
-        activityData.amount = txData.receives[0]?.amount;
-        activityData.symbol = getTokenSymbol(txData.receives[0]?.token_id);
-        activityData.transactionLabel = TransactionLabelEnum.Received;
+    const isGasToken =
+      !(txData.sends.length > 0 && txData.receives.length > 0) &&
+      isDefined(txData.tx) &&
+      txData.tx.value > 0 &&
+      txData.cate_id === null &&
+      txData.token_approve === null;
+    const isGasTokenSent = publicKeyHash.toLowerCase() === txData.tx?.from_addr?.toLowerCase();
+
+    const { type, label } = getTransactionType(txData, isGasToken, isGasTokenSent);
+    activityData.type = type;
+    activityData.label = label;
+
+    switch (type) {
+      case TransactionTypeEnum.Send: {
+        if (isGasToken) {
+          activityData.symbol = chainName;
+          activityData.amount = txData.tx?.value;
+        } else {
+          activityData.amount = txData.sends[0]?.amount;
+          activityData.tokenId = txData.sends[0]?.token_id;
+          activityData.symbol = getTokenSymbol(txData.sends[0]?.token_id);
+          activityData.isCollectible = isDefined(userTokensMetadata[txData.sends[0]?.token_id].thumbnail_url);
+        }
+        break;
       }
-    } else {
-      activityData.symbol = chainName;
-      activityData.amount = txData.tx?.value;
-      if (publicKeyHash.toLowerCase() === txData.tx?.from_addr?.toLowerCase()) {
-        activityData.transactionLabel = TransactionLabelEnum.Send;
-      } else {
-        activityData.transactionLabel = TransactionLabelEnum.Received;
+      case TransactionTypeEnum.Receive: {
+        if (isGasToken) {
+          activityData.symbol = chainName;
+          activityData.amount = txData.tx?.value;
+        } else {
+          activityData.tokenId = txData.receives[0]?.token_id;
+          activityData.amount = txData.receives[0]?.amount;
+          activityData.symbol = getTokenSymbol(txData.receives[0]?.token_id);
+          activityData.isCollectible = isDefined(userTokensMetadata[txData.receives[0]?.token_id].thumbnail_url);
+        }
+        break;
+      }
+      case TransactionTypeEnum.Interaction:
+      case TransactionTypeEnum.ContractInteraction: {
+        activityData.projectName = isDefined(txData.project_id) ? getProjectName(txData.project_id) : '';
+        activityData.transfer = {
+          sends: [
+            ...(txData.sends.map(item => ({
+              amount: item.amount,
+              symbol: getTokenSymbol(item.token_id)
+            })) ?? [])
+          ],
+          receives: [
+            ...(txData.receives.map(item => ({
+              amount: item.amount,
+              symbol: getTokenSymbol(item.token_id)
+            })) ?? [])
```

Comment:
why do you need `?? []` fallback here?  due to types - `sends` and `receives` should always be arrays